autopush-common/src/db/bigtable/bigtable_client/mod.rs

```diff
@@ -89,6 +87,63 @@ fn timestamp_filter() -> Result<data::RowFilter, error::BigTableError> {
     Ok(timestamp_filter)
 }
 
+/// Escape bytes for RE values
+///
+/// Based off google-re2/perl's quotemeta function
+fn escape_bytes(bytes: &[u8]) -> Vec<u8> {
+    let mut vec = Vec::with_capacity(bytes.len() * 2);
+    for &b in bytes {
+        if !b.is_ascii_alphanumeric() && b != b'_' && (b & 128) == 0 {
+            if b == b'\0' {
+                // Special handling for null: Note that this special handling
+                // is not strictly required for RE2, but this quoting is
+                // required for other regexp libraries such as PCRE.
+                // Can't use "\\0" since the next character might be a digit.
+                vec.extend("\\x00".as_bytes());
+                continue;
+            }
+            vec.push(b'\\');
+        }
+        vec.push(b);
+    }
+    vec
+}
+
+/// Return a RowFilter limiting to a match of the specified `version`'s column
+/// value
+fn version_filter(version: &Uuid) -> data::RowFilter {
+    let mut router_filter_chain = RowFilter_Chain::default();
+    let mut filter_set: RepeatedField<RowFilter> = RepeatedField::default();
+
+    let mut family_filter = data::RowFilter::default();
+    family_filter.set_family_name_regex_filter(format!("^{ROUTER_FAMILY}$"));
+
+    let mut cq_filter = data::RowFilter::default();
+    cq_filter.set_column_qualifier_regex_filter("^version$".as_bytes().to_vec());
+
+    let mut value_filter = data::RowFilter::default();
+    value_filter.set_value_regex_filter(escape_bytes(version.as_bytes()));
+
+    filter_set.push(family_filter);
+    filter_set.push(cq_filter);
+    filter_set.push(value_filter);
+
+    router_filter_chain.set_filters(filter_set);
+    let mut router_filter = RowFilter::default();
+    router_filter.set_chain(router_filter_chain);
+    router_filter
+}
+
+/// Return a newly generated `version` column `Cell`
+fn new_version_cell(timestamp: SystemTime) -> cell::Cell {
```

Comment:
will the timestamp ever be anything other than `std::time::SystemTime::now() + Duration::from_secs(MAX_ROUTER_TTL)`?

Reply:
Nope but I'd rather this just 100% match with other cells in the same operation, whereas having this function make its own `SystemTime` could potentially make them differ by a ms or two.